pkg/cli/mirror/mirror.go

```diff
@@ -704,3 +696,17 @@ func (o *MirrorOptions) checkErr(err error, acceptableErr func(error) bool) erro
 
 	return nil
 }
+
+func writeMappingFile(mappingPath string, mapping image.TypedImageMapping) error {
+	path := filepath.Clean(mappingPath)
+	mappingFile, err := os.Create(path)
+	if err != nil {
+		return err
+	}
+	defer mappingFile.Close()
+	klog.Infof("Writing image mapping to %s", mappingPath)
+	if err := image.WriteImageMapping(mapping, mappingFile); err != nil {
+		return err
+	}
+	return mappingFile.Sync()
```

Comment:
@dinhxuanvu Wanted to call out this change. The defer statements were getting pretty illegible so I changed to this pattern, but this may be overkill (performance-wise). I figured since it is only running one time and the point of `dry-run` is to write these files it may be a good trade-off. WDYT?